Sparkle/SUTouchBarButtonGroup.m

```diff
@@ -0,0 +1,60 @@
+//
+//  SUTouchBarButtonGroup.m
+//  Sparkle
+//
+//  Created by Yuxin Wang on 05/01/2017.
+//  Copyright © 2017 Sparkle Project. All rights reserved.
+//
+
+#import "SUTouchBarButtonGroup.h"
+
+@implementation SUTouchBarButtonGroup
+
+@synthesize buttons = _buttons;
+
+- (instancetype)initByReferencingButtons:(NSArray<NSButton *> *)buttons
```

Comment:
It may be looks like
```
- (instancetype)initByReferencingDefaultButton:(NSButton *)defaultButton alternateButton:(nullable NSButton *)alternateButton otherButton:(nullable NSButton *)otherButton;
```
But it conflicts with the property `buttons`. It may need to three properties for holding three buttons.
The view places buttons from right to left, just like `NSAlert`. It is designed to rely on indices. Also the view does not care about the purpose of the buttons, it cares about layout only. I think it is good to keep the button array.